### DISCUSSION TITLE
`build_actions` and `export_actions` are not work

### DISCUSSION THREAD

**Discussion by rieval** [framesurge/perseus#155] [Q&A] [answered]
Here is my code:
```rust
// plug.rs
pub fn new_plug<G: perseus::Html>() -> perseus::plugins::Plugin<G, ()> {
    use perseus::plugins::{empty_control_actions_registrar, Plugin, PluginAction, PluginEnv};
    Plugin::new(
        "new-plug",
        |mut actions| {
            actions
                .build_actions
                .before_build
                .register_plugin("new-plug", |_, _| {});
            actions
                .export_actions
                .before_export
                .register_plugin("new-plug", |_, _| {});
            actions
        },
        empty_control_actions_registrar,
        PluginEnv::Server,
    )
}

// lib.rs
#[perseus::main(perseus_axum::dflt_server)]
pub fn main<G: Html>() -> PerseusApp<G> {
    PerseusApp::new()
        .error_pages(crate::error_pages::get_error_pages)
        .plugins(Plugins::new().plugin(new_plug, ()))
// snip
```

When I tired to run `perseus serve`, it seems fine.
![image](https://user-images.githubusercontent.com/22883406/178460864-8add452c-3530-49a2-bb48-f90a85412e4c.png)

However, the `pkg` folder is missing.
![image](https://user-images.githubusercontent.com/22883406/178461270-b98a211d-a810-4ff0-8199-5f424869c46a.png)

**Comment by rieval** [comment, marked as answer]
Oh sorry sir, I'm forgot to add `#[cfg(not(target_arch = "wasm32"))]` flag 😂

**Reply by arctic-hen7** [reply]
Is it working now?

**Reply by rieval** [reply]
> Is it working now?

Yes. 😀

**Reply by arctic-hen7** [reply]
Great! Could anything be clearer in the docs do you think? I'm always looking for pain points to fix!

**Reply by rieval** [reply]
> Great! Could anything be clearer in the docs do you think? I'm always looking for pain points to fix!

I can't tell the difference in docs between the `next` and `v0.4.x` versions. Usually, I learned the relevant content by reading its examples directly. Thankfully, `perseus` has a lot of clear examples 😀

**Reply by arctic-hen7** [reply]
Yeah, that can be a problem. I'll think about how that might be made clearer. Right now though, I've just released beta 4, so the `next` and `0.4.x` docs are now identical. (For the next day or so at least!)

**Reply by rieval** [reply]
> Yeah, that can be a problem. I'll think about how that might be made clearer. Right now though, I've just released beta 4, so the `next` and `0.4.x` docs are now identical. (For the next day or so at least!)

Great work!
